pkg/agent/controller/noderoute/node_route_controller.go

```diff
@@ -126,35 +129,104 @@ func (c *Controller) enqueueNode(obj interface{}) {
 	}
 }
 
+// listRoutes reads all the routes for the gateway and returns them as a map with the destination
+// subnet as the key.
+func (c *Controller) listRoutes() (routes map[string]*netlink.Route, err error) {
+	routes = make(map[string]*netlink.Route)
+	routeList, err := netlink.RouteList(c.gatewayLink, netlink.FAMILY_V4)
+	if err != nil {
+		return routes, err
+	}
+	for _, route := range routeList {
+		routes[route.Dst.String()] = &route
+	}
+	return routes, nil
+}
+
+// removeStaleGatewayRoutes removes all the gateway routes which no longer correspond to a Node in
+// the cluster. If the antrea agent restarts and Nodes have left the cluster, this function will
+// take care of removing routes which are no longer valid.
+func (c *Controller) removeStaleGatewayRoutes() error {
```

Comment:
I merged the IPSec PR. Seems you should remove IPSec tunnel ports as well. Maybe call deleteNodeRoute for the deleted Nodes.

Reply:
will update the PR